pkg/controller/secret/secret_controller.go

```diff
@@ -433,6 +436,16 @@ func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileSecret) getClusterID() string {
+	var version configv1.ClusterVersion
+	err := r.client.Get(context.TODO(), client.ObjectKey{Name: "version"}, &version)
+	if err != nil {
+		log.Error(err, "Error reading cluster id.")
```

Comment:
Can we return this as an `error`, and then log it in the main loop (with the log with variables)?

Reply:
Sure, I don't have any preference about where we hide the error ;)